Source/WebCore/css/CSSCustomPropertyValue.cpp

```diff
@@ -110,30 +110,38 @@ String CSSCustomPropertyValue::customCSSText() const
                     builder.append(separator);
                 builder.append(serializeSyntaxValue(syntaxValue));
             }
-            m_stringValue = builder.toString();
+            return builder.toString();
         });
-    }
-    return m_stringValue;
+    };
+
+    if (m_cachedCSSText.isNull())
+        m_cachedCSSText = serialize();
+
+    return m_cachedCSSText;
 }
 
-Vector<CSSParserToken> CSSCustomPropertyValue::tokens() const
+const Vector<CSSParserToken>& CSSCustomPropertyValue::tokens() const
 {
-    Vector<CSSParserToken> result;
-    WTF::switchOn(m_value, [&](const std::monostate&) {
+    static NeverDestroyed<Vector<CSSParserToken>> emptyTokens;
+
+    return WTF::switchOn(m_value, [&](const std::monostate&) -> const Vector<CSSParserToken>& {
```

Comment:
Are all the explicit `const Vector<CSSParserToken>&` return value types absolutely necessary?

Reply:
yeah, sadly it deduces Vector<CSSParserToken> otherwise